kolibri/utils/server.py

```diff
@@ -403,29 +406,41 @@ def ENTER(self):
         self.process_pid = os.getpid()
 
 
-class RestartPlugin(Monitor):
+class CommandPlugin(Monitor):
     def __init__(self, bus):
         self.mtime = self.get_mtime()
         Monitor.__init__(self, bus, self.run, 1)
 
     def get_mtime(self):
         try:
-            return os.stat(RESET_FLAG).st_mtime
+            return os.stat(COMMAND_FLAG).st_mtime
         except OSError:
             return 0
 
     def run(self):
         mtime = self.get_mtime()
         if mtime > self.mtime:
             # The file has been deleted or modified.
-            self.bus.log("Restarting server.")
-            self.thread.cancel()
-            self.bus.restart()
+            with open(COMMAND_FLAG, "r") as f:
+                try:
+                    command = f.read().strip()
+                except (IOError, OSError):
```

Comment:
I like the specificity of error handling here.  Could you add a comment about when these errors happen?  I'm assuming it's when the flag file doesn't exist?

Reply:
Yes, or the file exists but can't be read - in Python 3, `IOError` and `OSError` were merged I believe, but in Python 2 we have to catch both.